linera-execution/src/runtime.rs

```diff
@@ -7,21 +7,24 @@ use crate::{
     resources::ResourceController,
     util::{ReceiverExt, UnboundedSenderExt},
     ApplicationCallOutcome, BaseRuntime, CallOutcome, CalleeContext, ContractRuntime,
-    ExecutionError, ExecutionOutcome, MessageContext, RawExecutionOutcome, ServiceRuntime,
-    SessionId, UserApplicationDescription, UserApplicationId, UserContractCode,
-    UserContractInstance, UserServiceInstance,
+    ExecutionError, ExecutionOutcome, FinalizeContext, MessageContext, RawExecutionOutcome,
+    ServiceRuntime, SessionId, UserApplicationDescription, UserApplicationId, UserContractInstance,
+    UserServiceInstance,
 };
 use custom_debug_derive::Debug;
 use linera_base::{
-    data_types::{Amount, ApplicationPermissions, ArithmeticError, BlockHeight, Timestamp},
+    data_types::{
+        Amount, ApplicationPermissions, ArithmeticError, BlockHeight, Resources, Timestamp,
+    },
     ensure,
     identifiers::{Account, ChainId, MessageId, Owner},
     ownership::ChainOwnership,
 };
 use linera_views::batch::Batch;
+use linked_hash_map::LinkedHashMap;
```

Comment:
I'm tempted to push back (again?) on adding this external library.
* External libraries add compile time and will need some security auditing down the road.
* This one says: "WARNING: THIS PROJECT IS IN MAINTENANCE MODE, DUE TO INSUFFICIENT MAINTAINER RESOURCES"
* Here, I believe we can just use `Vec` and `HashMap`.

Reply:
Right! I had forgotten why we decided not to use it last time :sweat_smile: